### PR TITLE
Rest

### PR DESCRIPTION
**Pull request by michelle-gu**
Considered adding a health check, but wasn't sure what to check -- I thought there might be more to work with in the later labs, and I can add one then

### CODE REVIEW

File: src/main/java/com/khoros/twitter/TwitterLabApplication.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import io.dropwizard.setup.Environment;
+
+public class TwitterLabApplication extends Application<TwitterLabConfiguration> {
+
+    public static void main(String[] args) throws Exception {
+        new TwitterLabApplication().run(args);
+    }
+
+    @Override
+    public void initialize(Bootstrap<TwitterLabConfiguration> bootstrap) {
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitter/TwitterLabApplication.java]: do you need this?

File: src/main/java/com/khoros/twitter/resources/GetTimelineResource.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import javax.ws.rs.core.MediaType;
+import java.util.ArrayList;
+import java.util.List;
+
+// This route will retrieve a list of latest tweets from the home timeline
+@Path("/api/1.0/twitter/timeline")
+@Produces(MediaType.APPLICATION_JSON)
+public class GetTimelineResource {
+
+    public GetTimelineResource() {
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitter/resources/GetTimelineResource.java]: do you actually need this?

File: src/main/java/com/khoros/twitter/resources/PostTweetResource.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+// This route will post the message to the Twitter account.
+@Path("/api/1.0/twitter/tweet")
+@Produces(MediaType.APPLICATION_JSON)
+public class PostTweetResource {
+
+    public PostTweetResource() {
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitter/resources/PostTweetResource.java]: same for this one

File: src/main/java/com/khoros/twitter/resources/PostTweetResource.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @POST
+    @Timed
+    public Response postTweet(Message message) {
+        // Post message
+        String text = message.getText();
+        Status status = null;
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitter/resources/PostTweetResource.java]: status is only used once, you can restructure the code better

File: src/main/java/com/khoros/twitter/resources/PostTweetResource.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        Message statusMessage = new Message();
+        try {
+            Twitter twitter = TwitterFactory.getSingleton();
+            if (text.length() > 0 || text.length() <= 280) {
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitter/resources/PostTweetResource.java]: constant for 280.

File: src/main/java/com/khoros/twitter/resources/GetTimelineResource.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @GET
+    @Timed
+    public Timeline getTimeline() {
+        List<Status> timeline = new ArrayList<Status>();
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitter/resources/GetTimelineResource.java]: what happens if it errors out?

File: src/main/java/com/khoros/twitter/resources/GetTimelineResource.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+@Path("/api/1.0/twitter/timeline")
+@Produces(MediaType.APPLICATION_JSON)
+public class GetTimelineResource {
+
+    public GetTimelineResource() {
+    }
+
+    @GET
+    @Timed
+    public Timeline getTimeline() {
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitter/resources/GetTimelineResource.java]: any reason why this returns Timeline and the other Resource endpoint returns a Response?

File: src/main/java/com/khoros/twitter/resources/PostTweetResource.java (hunk 5 above)
Comment (kev-chan) [reply src/main/java/com/khoros/twitter/resources/PostTweetResource.java]: look into your logic

File: src/main/java/com/khoros/twitter/TwitterLabApplication.java (hunk 1 above)
Comment (michelle-gu) [reply src/main/java/com/khoros/twitter/TwitterLabApplication.java]: initialize? i think it needs it to extend Application

**Comment by kev-chan** [comment]
is there more useful info that can be returned on the postTweet endpoint?

**Pull request merged**